inst/tinytest/test_a_delayed_1.R

```diff
@@ -0,0 +1,102 @@
+# ================================================================
+# Context of test_delayed_1.R, test_delayed_2.R, etc:
+#
+# Since we need mutable references in the DAG data structure, we use R
+# reference classes, which have their own semantics and which may not be
+# familiar to everyone. Integration with futures adds complexity. And lastly,
+# testing with TileDB Cloud UDFs adds distributed-computing complexity.
+#
+# These things make it unclear where to *start* -- not only for initial
+# development of this package, but for any future deep rework of this package:
+# * Porting to another language
+# * Dealing with a disruptive upgrade of some dependency
+# * Replacing one dependent package with another
+# * Refactoring for whatever reason.
+# -- for any of those, people will likely need to recapitulate the development
+# sequencing.
+#
+# For these reasons, these test are split up, and -- since tests are listed and
+# run in alphabetical order -- their filenames are chosen to reflect the
+# chronological development order.
+# * Test out the delayed function, and the Delayed, Node, and DAG reference
+#   classes syntactically. Make sure sure mutability and references are
+#   validated first. Do this with purely local functions which do things
+#   like sleep two seconds and return the sum or product of two arguments.
+# * Introduce futures and test our understanding of that library iteratively.
+# * Then test attaching futures to the DAG computation.
+# * Then add in remote execution of UDFs on TileDB Cloud without futures.
+# * Then add in remote execution of UDFs on TileDB Cloud with futures.
+# * Etc.
+# ================================================================
+
+library(tiledbcloud)
+library(tinytest)
+
+# ----------------------------------------------------------------
+a <- delayed(function(...){sum(...)}, args=list(1,2))
+expect_equal(a$compute_sequentially(), 3)
+
+# ----------------------------------------------------------------
+a <- delayed(function(...){sum(...)}, args=list(1,2))
+a$set_args(list(3, 4))
+expect_equal(a$compute_sequentially(), 7)
+
+# ----------------------------------------------------------------
+a <- delayed(function(...){sum(...)})
+a$set_args(list(5, 6))
+expect_equal(a$compute_sequentially(), 11)
+
+# ----------------------------------------------------------------
+a = delayed(function() { Sys.sleep(0.1); 3 })
```

Comment:
nag: might be nicer to consistently use `<-` ?

Reply:
thanks @eddelbuettel ! i've been coding with `=` for assignment for 30 years now and i will occasionally make this error as I'm still new to R -- thanks for catching it!! i'd love to think this'll be the last time i ever make this mistake but honestly i doubt it, so thanks for being a sharp second pair of eyes! :)